check/check.go

```diff
@@ -339,5 +349,4 @@ func exitWithError(err error) {
 	fmt.Fprintf(os.Stderr, "\n%v\n", err)
 	// flush before exit non-zero
 	glog.Flush()
-	os.Exit(1)
```

Comment:
Why have you removed this? `exitWithError()` is supposed to terminate the program with a non-zero error code.

Reply:
I removed this because I thought that we won't want to terminate the program if there is one wrong command no?
Because then just one wrong thing in 1 test out of 100 will break it.
Let say there is a use of command "XXX" that doesn't exist on the running environment could break the program instead of running it, and in the output, say that one test failed because command "XXX" is not exist

Reply:
The clue is in the name `exitWithError` which does suggest that the calling code should only call this when they want to exit. And it really should exit. 

More generally, if you want to change behaviour you should at the very least call it out in the conversation, because it would have been very easy for me to miss this.